ansible/roles/lib_aos_modules/library/aos_rhsm_repos.py

```diff
@@ -0,0 +1,202 @@
+  required: false
+  default: present
+  aliases: []
+query:
+  choices: ['all', 'enabled', 'disabled']
+  description:
+  - The query type when looking at repositories.
+  required: false
+  default: present
+  aliases: []
+'''
+EXAMPLES = '''
+# perform a list on the enabled repos
+- aos_rhsm_repos:
+    state: list
+    query: enabled
+  register: repos
+
+# Set repositories 2 repositories and disable the others.
+- aos_rhsm_repos:
+    state: present
+    enabled:
+    - rhel-7-server-rpms
+    - rhel-7-server-extras-rpms
+    disabled:
+    - '*'
+'''
+import re
+
+RHSM_BIN = '/usr/sbin/subscription-manager'
+
+class RHSMError(Exception):
+    """rhsm error"""
+    pass
+
+class RHSMRepos(object):
+    """simple wrapper class for rhsm repos"""
+    def __init__(self, module):
+        self.module = module
+        self.enabled = module.params['enabled']
+        self.disabled = module.params['disabled']
+
+    def set_repositories(self):
+        """set the enabled and disabled repositories"""
+        cmd = [RHSM_BIN, 'repos']
+        if self.disabled:
+            cmd.extend(['--disable=%s' % dis for dis in self.disabled])
+        # enabled must come last as disable will undo the enabled.
+        if self.enabled:
+            cmd.extend(['--enable=%s' % ena for ena in self.enabled])
+
+        rcode, out, err = self.module.run_command(cmd)
+        if rcode != 0:
+            raise RHSMError('error setting repositories. rc=[%s] cmd=[%s] err=[%s]' % (rcode, cmd, err))
+
+        return out
+
+    @staticmethod
+    def parse_repos(rinput):
+        """parse the repo output"""
+        all_repos = []
+        # split rinput on double new lines
+        repos = re.split('\n\n', rinput)
+        for idx, repo in enumerate(repos):
+            repo.strip()
+            if not repo:
+                continue
+
+            repo = repo.split('\n')
+
+            # remove the first 3 lines
+            if idx == 0:
+                repo = repo[3:]
+
+            if len(repo) != 4:
+                raise RHSMError('error parsing repositories. repo=%s' % repo)
+
+            all_repos.append({'id': repo[0].split('ID:')[1].strip(),
+                              'name': repo[1].split('Name:')[1].strip(),
+                              'url': repo[2].split('URL:')[1].strip(),
+                              'enabled': int(repo[3].split('Enabled:')[1].strip()) == 1
+                             })
+
+        return all_repos
+
+
+    def query_repos(self, qtype='list'):
+        """query for repos"""
+
+        cmd = [RHSM_BIN, 'repos']
+
+        if qtype == 'enabled':
+            cmd.append('--list-enabled')
+        elif qtype == 'disabled':
+            cmd.append('--list-disabled')
+        else:
+            cmd.append('--list')
+
+        rcode, out, err = self.module.run_command(cmd)
+        if rcode != 0:
+            raise RHSMError('error querying repositories. rc=[%s] cmd=[%s] err=[%s]' % (rcode, cmd, err))
```

Comment:
Should this be a `self.module.fail_json` instead?

-or- should we wrap the caller in a try/except and do a `self.module.fail_json`  ?